packages/@aws-cdk/aws-applicationsignals-alpha/lib/slo/metric.ts

```diff
@@ -0,0 +1,357 @@
+import { ComparisonOperator, MetricStatistic, MetricType } from './constants';
+import { KeyAttributes } from './keyAttributes';
+
+/**
+ * Interface for metric dimension
+ */
+export interface MetricDimension {
+    /**
+     * Name of the dimension
+     *
+     * @required
+     */
+    readonly name: string;
+
+    /**
+     * Value of the dimension
+     *
+     * @required
+     */
+    readonly value: string;
+}
+
+/**
+ * Properties for Application Signals metrics
+ */
+export interface ApplicationSignalsMetricProps {
+    /**
+     * Type of the metric
+     *
+     * @required
+     */
+    readonly metricType: MetricType;
+
+    /**
+     * Key attributes for the service
+     *
+     * @required
+     */
+    readonly keyAttributes: KeyAttributes;
+
+    /**
+     * Operation name
+     *
+     * @default - undefined
+     */
+    readonly operationName?: string;
+
+    /**
+     * Period in seconds
```

Comment:
If it will be internal function, so it is Ok to use union, but if the plan is to export it to function, so replace the union type with a parent type for all of these concrete types.